lib/spack/spack/ci.py

```diff
@@ -514,6 +515,63 @@ def format_job_needs(phase_name, strip_compilers, dep_jobs,
     return needs_list
 
 
+def get_change_revisions():
+    """If this is a git repo get the revisions to use when checking
+    for changed packages and spack core modules."""
+    git_dir = os.path.join(spack.paths.prefix, '.git')
+    if os.path.exists(git_dir) and os.path.isdir(git_dir):
+        # TODO: This will only find changed packages from the last
+        # TODO: commit.  While this may work for single merge commits
+        # TODO: when merging the topic branch into the base, it will
+        # TODO: require more thought outside of that narrow case.
+        return 'HEAD^', 'HEAD'
+    return None, None
+
+
+def compute_affected_packages(rev1='HEAD^', rev2='HEAD'):
+    """Determine which packages were added, removed or changed
+    between rev1 and rev2, and return the names as a set"""
+    return pkg_cmd.get_all_package_diffs('ARC', rev1=rev1, rev2=rev2)
+
+
+def get_spec_filter_list(env, affected_pkgs, dependencies=True, dependents=True):
+    """Given a list of package names, and assuming an active and
+       concretized environment, return a set of concrete specs from
+       the environment corresponding to any of the affected pkgs (or
+       optionally to any of their dependencies/dependents).
+
+    Arguments:
+
+        env (spack.environment.Environment): Active concrete environment
+        affected_pkgs (List[str]): Affected package names
+        dependencies (bool): Include dependencies of affected packages
+        dependents (bool): Include dependents of affected pacakges
+
+    Returns:
+
+        A list of concrete specs from the active environment including
+        those associated with affected packages, and possible their
+        dependencies and dependents as well.
+    """
+    affected_specs = set()
+    all_concrete_specs = env.all_specs()
+    tty.debug('All concrete environment specs:')
+    for s in all_concrete_specs:
+        tty.debug('  {0}/{1}'.format(s.name, s.dag_hash()[:7]))
+    for pkg in affected_pkgs:
+        pkg_spec = spack.spec.Spec(pkg)
+        env_matches = [s for s in all_concrete_specs if s.satisfies(pkg_spec)]
+        for match in env_matches:
+            affected_specs.add(match)
+            if dependencies:
+                for d in match.traverse(direction='children', root=False):
+                    affected_specs.add(d)
```

Comment:
I think we can do:
```suggestion
                affected_specs.update(match.traverse(direction='children', root=False))
```
and similarly below.

Reply:
Sure, will do.  Is it about performance or code style (or number of lines)?

Reply:
Mainly personal preference on code style, not sure about performance but it won't be worse I guess. It's the "set" version of using:
```python
list.extend(sequence)
```
instead of:
```python
for item in sequence:
   list.append(item)
```